Docs: Part 2 of the rocSOLVER documentation refresh

amd-jnovotny opened the pull request:
This PR continues the rocSOLVER docs refresh with copy edit and formatting corrections for the how-to and non-autogenerated sections of the API reference docs. It also removes a now-unused file.

amd-jnovotny committed:
Docs: Part 2 of the rocSOLVER docs refresh

tfalders commented on docs/howto/contribute.rst:
```diff
@@ -1,25 +1,21 @@
 .. meta::
-  :description: rocSOLVER documentation and API reference library
-  :keywords: rocSOLVER, ROCm, API, documentation
+  :description: rocSOLVER contribution guide,
```
Extraneous comma

tfalders commented on docs/reference/helpers.rst:
```diff
@@ -11,9 +11,9 @@ rocSOLVER library and logging functions
-* :ref:`api_logging`. These functions control the :ref:`logging-label` capabilities.
+* :ref:`lib_info`: Return information about the library version.
+* :ref:`algo_select`: Select different algorithm modes of certain APIs.
+* :ref:`api_logging`. Control the :ref:`logging-label` capabilities.
```
No colon?

tfalders commented on docs/reference/intro.rst:
```diff
@@ -1,23 +1,20 @@
-The following tables summarize the functionality implemented for the different supported precisions in rocSOLVER's latest release.
-All LAPACK and LAPACK-like main functions include *_batched* and *_strided_batched* versions. For a complete description of the listed
-routines, please see the corresponding reference guides.
+   The rocSOLVER library is in the early stages of active development, with new features are being continuously added and documented.
```
Extraneous "are"

tfalders reviewed (approved):
Looks good! I just have a few minor comments

tfalders commented:
There also seems to be a conflict that needs to be resolved

amd-jnovotny committed:
Merge branch 'develop' into rocsolver-docs-refresh-part2-develop

amd-jnovotny committed:
Incorporate feedback from SME review

amd-jnovotny commented:
@tfalders Thanks again for the feedback. I've added your suggestions to the new commit and fixed up the merge conflict at the same time. It looks like a very old (but very minor) PR got merged recently, but it was easy to resolve the conflict.
